mlrun/datastore/datastore_profile.py

```diff
@@ -193,6 +196,10 @@ class DatastoreProfileKafkaSource(DatastoreProfile):
     kwargs_public: typing.Optional[dict]
     kwargs_private: typing.Optional[dict]
 
+    def get_topic(self):
+        topics = [self.topics] if isinstance(self.topics, str) else self.topics
+        return topics[0] if topics else None
```

Comment:
Why did you choose the first topic? It seems arbitrary and can cause issues.

Missing type hint:

```suggestion
    def get_topic(self) -> typing.Optional[str]:
        topics = [self.topics] if isinstance(self.topics, str) else self.topics
        return topics[0] if topics else None
```

But actually `topics` is guaranteed to be non-empty, so the last `if else` is redundant, isn't it?

Reply:
It's not guaranteed to be non empty. User can provide an empty list or empty string to the DataStore profile
As for the first topic - this is how its used down the road. We will have to review the multiple topics when refactoring the Kafka interfaces